### DISCUSSION TITLE
Atlas K8S Operator: Checksum mismatch -- why??

### DISCUSSION THREAD

**Discussion by jimsnab** [ariga/atlas#2809] [Q&A]
I am starting out with Atlas and have it set up in Kubernetes as described in [Versioned Migrations](https://atlasgo.io/integrations/kubernetes/operator#versioned-migrations). I added an initial table and it worked. Then I added a second table, decided I wanted to rearrange and rename the source files, tried to reset and start over, and now I am stuck.

I have...
* run atlas migrate hash to make sure `atlas.sum` has a proper checksum, of course
* deleted the entire postgres database on k8s and made a new one
* inspected the `migration-dir` configmap and see that it matches the source code
* successfully manually run `atlas migrate apply` from my local machine to the k8s postgres

Yet the operator produces this:

```
Events:                                                                                                                                           
  Type     Reason  Age   From                       Message                                                                                       
  ----     ------  ----  ----                       -------                                                                                       
  Warning  Error   6s    atlasmigration-controller  Error: checksum mismatch    
```

Is the operator latched on the first checksum? I thought this was maintained in the database that I deleted; where else would it be stored?

Is there a way to turn on a more verbose operator log?

**Comment by giautm** [comment]
Hello, there is a "snapshot" of your directory in k8s' secret. Can you please try to delete it too? And see what happen next? Some additional question:
- What's the version of atlas-operator?
- What's your OS when creating the configmap for Migration Directory?

Thank you, I'll back to answer your question soon (if any)

**Comment by jimsnab** [comment]
Thanks for the reply.

The issue was this. When the files are placed in the config map, the new line at the end of the file is lost, and therefore the checksums compute differently. I removed the last \n and it is working.
